src/libraries/Directory.Build.targets

```diff
@@ -184,10 +184,12 @@
   <!-- Define this target to override the workaround in arcade as we don't need it for our pkgprojs -->
   <Target Name="InitializeStandardNuspecProperties" />
 
-  <Target Name="AddSkipGetTargetFrameworkToProjectReferences" Condition="'@(ProjectReference)' != ''">
+  <Target Name="UpdateProjectReferencesWithAttributes" Condition="'@(ProjectReference)' != ''">
     <ItemGroup>
       <ProjectReference>
         <SkipGetTargetFrameworkProperties>true</SkipGetTargetFrameworkProperties>
+        <AdditionalProperties Condition="'%(Filename)' == 'System.Private.CoreLib' and '$(RuntimeFlavor)' == 'CoreCLR'">Configuration=$(CoreCLRConfiguration);TargetFramework=$(NetCoreAppCurrent)</AdditionalProperties>
```

Comment:
When building the repository I believe this can cause double build of System.Private.CoreLib.  Since it will have already been built without these global properties.  It can also result in race conditions if CoreLib is being built in the same parallel project graph as traversal.

Do we need to specify TargetFramework?  It looks like that's already set in S.P.C.  Can't we just UndefineProperties TargetFramework instead so that it doesn't flow?

Can we avoid setting Configuration if it is already set?

Can we append to AdditionalProperties when setting?

Reply:
@viktorhofer did you see this?

Reply:
> @ViktorHofer did you see this?

@ericstj yeah but forgot to respond, sorry.

> Do we need to specify TargetFramework? It looks like that's already set in S.P.C. Can't we just UndefineProperties TargetFramework instead so that it doesn't flow?

Sure if that's possible. How would I prevent the property from flowing? Currently the TargetFramework Sdk controls that: https://github.com/dotnet/arcade/blob/master/src/Microsoft.DotNet.Build.Tasks.TargetFramework.Sdk/src/build/Microsoft.DotNet.Build.Tasks.TargetFramework.Sdk.targets#L59. Do we need a hook in that sdk?

Reply:
> Can we avoid setting Configuration if it is already set?

In which case would configuration already be set? We are not flowing that property.

Reply:
> prevent the property from flowing

You can set the UndefineProperties metadata on the ProjectReference.  If the config system doesn't honor it we might have to add a switch.

Global properties automatically flow.  So Configuration will always be either 1) whatever the default is (eg: debug) or 2) specified as a global property by build.cmd|sh

In general we need to be *extra* careful about global property hygine.